Adapt the contributions guide for Sharg (#49)
seqan/sharg-parser: CONTRIBUTING.md

# Contributing {#about_contributing}

> First of all, thanks for wanting to contribute to the Sharg-parser from the SeqAn team! Community is important to us and we strive to maintain a great
> culture and atmosphere. Please have a look at our [Code of Conduct](\ref about_code_of_conduct).

Reviewer: ```suggestion
First of all, thanks for wanting to contribute to the SeqAn's Sharg-parser! Community is important to us and we strive 
to maintain a great culture and atmosphere. Please have a look at our [Code of Conduct](\ref about_code_of_conduct).
```

## Creating an issue

> Go to https://github.com/seqan/sharg-parser/issues and check if someone else has already reported the same issue. If not,
> create a new issue and follow the template.

Reviewer: ```suggestion
Go to https://github.com/seqan/sharg-parser/issues and check if someone else has already reported the same issue. If 
not, create a new issue and follow the template.
```

### Request first review

After opening a PR it goes through the review process. This is a two-step process, first a regular member of the team
needs to approve your changes, then one of the SeqAn core members needs to approve and merge it. If you

> are a collaborator of the project and know the SeqAn team member most suitable for the review you can request a review

Reviewer: ```suggestion
are a collaborator of the project and know the SeqAn team member most suitable for the review, you can request a review
```

yourself. Otherwise, wait for a reviewer to be assigned to the PR. Do not request reviews from a core member at this
point. Always only request review from one person at a time.

### Resolve comments

If you have not contributed to SeqAn before, you will likely receive a lot of comments on the style of your code. This
is not a sign that we don't appreciate your contribution, but we have high code quality standards and everything

> must conform to our [style guide](https://github.com/seqan/seqan3/wiki#library-coding-guide). In the future we hope to automate this step.

Reviewer: ```suggestion
must conform to our [style guide](https://github.com/seqan/seqan3/wiki#library-coding-guide). In the future we hope 
to automate this step.
```

(**please do not mark another person's comments as resolved**). They will then add a second review or approve your
changes. Repeat the last steps until you get approval.

### Request review by a SeqAn core member

> Once your changes are approved by one reviewer, request a second review from a core member (or wait for it to be assigned).
> They will go through everything briefly again and make sure that nothing escaped the first reviewer's notice.
> Ultimately, they will approve your changes and merge your branch.

Reviewer: ```suggestion
Once your changes are approved by one reviewer, request a second review from a core member (or wait for assignment).
They will go through everything briefly again and make sure that nothing escaped the first reviewer's notice.
Ultimately, they will approve your changes and merge your branch.
```
